test/Regex/characterclass_with_range.js

```diff
@@ -0,0 +1,163 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+let re = /^[\s-a-z]$/;
+let reIgnoreCase = /^[\s-a-z]$/i;
+let reUnicode = /^[\s-z]$/u;
+let reNoCharClass = /^[a-c-z]$/;
+
+var tests = [
+    /*No Flag RegExp Tests begin*/
+    {
+        name : "Ensure 'a-z' not counted as range",
+        body : function () 
+        {
+            assert.isFalse(re.test("b"));
+        }
+    },
+    {
+        name : "Ensure 'a' included in set",
+        body : function () 
+        {
+            assert.isTrue(re.test("a"));
+        }
+    },
+    {
+        name : "Ensure ' ' included in set",
+        body : function () 
+        {
+            assert.isTrue(re.test(" "));
+        }
+    },
+    {
+        name : "Ensure 'z' included in set",
+        body : function () 
+        {
+            assert.isTrue(re.test("z"));
+        }
+    },
+    {
+        name : "Ensure '\t' included in set",
+        body : function () 
+        {
+            assert.isTrue(re.test("\t"));

```

Comment:
>\t [](start = 35, length = 2)

it would be a good idea to test one of the non-ASCII whitespaces included in \s.  (see StandardChars.cpp(192): `StandardChars<char16>::whitespaceStr`)

Reply:
One of either: paragraph separator (U+2028) or line separator (U+2029) might be a good case here, assuming we already handle them correctly. [https://en.wikipedia.org/wiki/Template:Whitespace_(Unicode](https://en.wikipedia.org/wiki/Template:Whitespace_(Unicode))

---
In reply to: [199985601](https://github.com/Microsoft/ChakraCore/pull/5418#discussion_r199985601) [](ancestors = 199985601)

Reply:
Added test cases for these characters